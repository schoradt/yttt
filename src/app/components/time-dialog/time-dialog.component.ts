import { Component } from '@angular/core';
import { MatCardModule } from '@angular/material/card';
import { MatButtonModule } from '@angular/material/button';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatInputModule } from '@angular/material/input';
import { DataService } from '../../services/data/data.service';
import {
  FormBuilder,
  FormControl,
  FormGroup,
  FormsModule,
  ReactiveFormsModule,
  Validators
} from '@angular/forms';
import { tap } from 'rxjs';
import { TimeTrack } from '../../model/time-track';

interface TimeTrackAddForm {
  ticket: FormControl<string>;
  description: FormControl<string | null>;
}

@Component({
  selector: 'yttt-time-dialog',
  imports: [
    MatCardModule,
    MatButtonModule,
    MatFormFieldModule,
    MatInputModule,
    FormsModule,
    ReactiveFormsModule
  ],
  templateUrl: './time-dialog.component.html',
  styleUrl: './time-dialog.component.scss'
})
export class TimeDialogComponent {
  form: FormGroup<TimeTrackAddForm>;

  constructor(
    private readonly dataService: DataService,
    private readonly formBuilder: FormBuilder
  ) {
    this.form = this.formBuilder.group({
      ticket: new FormControl('', {
        nonNullable: true,
        validators: [
          Validators.required,
          Validators.minLength(2),
          Validators.maxLength(20)
        ]
      }),
      description: new FormControl('', {})
    });
  }

  submit() {
    const values = this.form.value;

    const ticket: TimeTrack = {
      startTime: new Date(),
      ticket: values.ticket || '',
      description: values.description || undefined
    };

    console.log(ticket);

    this.dataService.storeTimeTrack(ticket).pipe(
      tap((track) => {
        console.log(track);
      })
    );
  }
}
