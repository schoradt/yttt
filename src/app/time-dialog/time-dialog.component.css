
mat-card {
  margin: 10px;
}

mat-card-header {
  border-bottom: solid;
}
