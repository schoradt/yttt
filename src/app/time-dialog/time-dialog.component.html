<mat-card appearance="outlined">
  <mat-card-header>
    <mat-card-title>Track Time</mat-card-title>
    <mat-card-subtitle>New entry</mat-card-subtitle>
  </mat-card-header>
  <mat-card-content>Simple card</mat-card-content>
  <mat-card-actions>
    <button mat-stroked-button>Add</button>
  </mat-card-actions>
</mat-card>

