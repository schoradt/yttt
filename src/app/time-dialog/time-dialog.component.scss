@use '@angular/material' as mat;

mat-card {
  margin: 10px;
}

mat-card-content {
  margin-top: 10px;
  margin-bottom: 10px;
}

.add-form {
  display: flex;
  flex: 1;
}

mat-form-field {
  flex: 1;
}
