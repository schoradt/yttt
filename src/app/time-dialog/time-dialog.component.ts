import { Component } from '@angular/core';
import { MatCardModule } from '@angular/material/card';
import { MatButtonModule } from '@angular/material/button';

@Component({
  selector: 'app-time-dialog',
  imports: [MatCardModule, MatButtonModule],
  templateUrl: './time-dialog.component.html',
  styleUrl: './time-dialog.component.css'
})
export class TimeDialogComponent {}
