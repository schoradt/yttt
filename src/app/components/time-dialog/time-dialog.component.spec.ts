import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TimeDialogComponent } from './time-dialog.component';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

describe('TimeDialogComponent', () => {
  let component: TimeDialogComponent;
  let fixture: ComponentFixture<TimeDialogComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [TimeDialogComponent, NoopAnimationsModule]
    }).compileComponents();

    fixture = TestBed.createComponent(TimeDialogComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should be rendered right', () => {
    expect(fixture).toMatchSnapshot();
  });
});
