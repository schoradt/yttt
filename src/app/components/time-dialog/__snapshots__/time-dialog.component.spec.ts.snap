// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeDialogComponent should be rendered right 1`] = `
<yttt-time-dialog
  dataService={[Function DataService]}
  form={[Function FormGroup]}
  formBuilder={[Function _FormBuilder]}
>
  <mat-card
    appearance="outlined"
    class="mat-mdc-card mdc-card mat-mdc-card-outlined mdc-card--outlined"
  >
    <mat-card-header
      class="mat-mdc-card-header"
    >
      <div
        class="mat-mdc-card-header-text"
      >
        <mat-card-title
          class="mat-mdc-card-title"
        >
          Track Time
        </mat-card-title>
        <mat-card-subtitle
          class="mat-mdc-card-subtitle"
        >
          New entry
        </mat-card-subtitle>
      </div>
    </mat-card-header>
    <mat-card-content
      class="mat-mdc-card-content"
    >
      <form
        class="form ng-untouched ng-pristine ng-invalid"
        novalidate=""
      >
        <div
          class="form-row"
        >
          <mat-form-field
            class="mat-mdc-form-field mat-mdc-form-field-type-mat-input mat-form-field-no-animations mat-form-field-appearance-fill mat-form-field-hide-placeholder mat-primary ng-untouched ng-pristine ng-invalid"
          >
            <div
              class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--filled"
            >
              <div
                class="mat-mdc-form-field-focus-overlay"
              />
              <div
                class="mat-mdc-form-field-flex"
              >
                <div
                  class="mat-mdc-form-field-infix"
                >
                  <label
                    class="mdc-floating-label mat-mdc-floating-label"
                    for="mat-input-a2"
                    id="mat-mdc-form-field-label-a2"
                    matformfieldfloatinglabel=""
                  >
                    <mat-label>
                      Ticket
                    </mat-label>
                    <span
                      aria-hidden="true"
                      class="mat-mdc-form-field-required-marker mdc-floating-label--required"
                    />
                  </label>
                  <input
                    aria-required="true"
                    class="mat-mdc-input-element mat-mdc-form-field-input-control mdc-text-field__input ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                    formcontrolname="ticket"
                    id="mat-input-a2"
                    matinput=""
                    required=""
                  />
                </div>
              </div>
              <div
                class="mdc-line-ripple mdc-line-ripple--deactivating"
                matformfieldlineripple=""
              />
            </div>
            <div
              class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
            >
              <div
                class="mat-mdc-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                style="opacity: 1; transform: translateY(0%);"
              >
                <div
                  class="mat-mdc-form-field-hint-spacer"
                />
              </div>
            </div>
          </mat-form-field>
        </div>
        <div
          class="form-row"
        >
          <mat-form-field
            class="mat-mdc-form-field mat-mdc-form-field-type-mat-input mat-form-field-no-animations mat-form-field-appearance-fill mat-form-field-hide-placeholder mat-primary ng-untouched ng-pristine ng-valid"
          >
            <div
              class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--filled"
            >
              <div
                class="mat-mdc-form-field-focus-overlay"
              />
              <div
                class="mat-mdc-form-field-flex"
              >
                <div
                  class="mat-mdc-form-field-infix"
                >
                  <label
                    class="mdc-floating-label mat-mdc-floating-label"
                    for="mat-input-a3"
                    id="mat-mdc-form-field-label-a3"
                    matformfieldfloatinglabel=""
                  >
                    <mat-label>
                      Work description
                    </mat-label>
                  </label>
                  <input
                    aria-invalid="false"
                    aria-required="false"
                    class="mat-mdc-input-element mat-mdc-form-field-input-control mdc-text-field__input ng-untouched ng-pristine ng-valid cdk-text-field-autofill-monitored"
                    formcontrolname="description"
                    id="mat-input-a3"
                    matinput=""
                  />
                </div>
              </div>
              <div
                class="mdc-line-ripple mdc-line-ripple--deactivating"
                matformfieldlineripple=""
              />
            </div>
            <div
              class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
            >
              <div
                class="mat-mdc-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                style="opacity: 1; transform: translateY(0%);"
              >
                <div
                  class="mat-mdc-form-field-hint-spacer"
                />
              </div>
            </div>
          </mat-form-field>
        </div>
      </form>
    </mat-card-content>
    <mat-card-actions
      align="end"
      class="mat-mdc-card-actions mdc-card__actions mat-mdc-card-actions-align-end"
    >
      <button
        class="mdc-button mdc-button--outlined mat-mdc-outlined-button _mat-animation-noopable mat-unthemed mat-mdc-button-base"
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
        mat-stroked-button=""
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-button__ripple"
        />
        <span
          class="mdc-button__label"
        >
          Add
        </span>
        <span
          class="mat-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </button>
    </mat-card-actions>
  </mat-card>
</yttt-time-dialog>
`;
